src/material-experimental/mdc-slider/slider.ts

```diff
@@ -0,0 +1,515 @@
+export const MAT_SLIDER_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => MatSlider),
+  multi: true
+};
+
+/** A simple change event emitted by the MatSlider component. */
+export class MatSliderChange {
+  /** The MatSlider that changed. */
+  source: MatSlider;
+
+  /** The new value of the source slider. */
+  value: number;
+}
+
+@Component({
+  moduleId: module.id,
+  selector: 'mat-slider',
+  templateUrl: 'slider.html',
+  styleUrls: ['slider.css'],
+  host: {
+    // The standard Angular Material slider has the capability to dynamically toggle
+    // whether tick markers should show or not. Therefore we need to make sure that
+    // the MDC slider foundation is able to render tick markers. We dynamically toggle
+    // them based on the specified "tickInterval" input.
+    'class': 'mat-mdc-slider mdc-slider mdc-slider--display-markers',
+    'role': 'slider',
+    'aria-orientation': 'horizontal',
+    // The tabindex if the slider turns disabled is managed by the MDC foundation which
+    // dynamically updates and restores the "tabindex" attribute.
+    '[attr.tabindex]': 'tabIndex || 0',
+    '[class.mdc-slider--discrete]': 'thumbLabel',
+    '[class.mat-slider-has-ticks]': 'tickInterval !== 0',
+    '[class.mat-slider-thumb-label-showing]': 'thumbLabel',
+    '[class.mat-slider-disabled]': 'disabled',
+    '[class.mat-primary]': 'color == "primary"',
+    '[class.mat-accent]': 'color == "accent"',
+    '[class.mat-warn]': 'color == "warn"',
+    '[class._mat-animation-noopable]': '_animationMode === "NoopAnimations"',
+    '(blur)': '_markAsTouched()',
+  },
+  exportAs: 'matSlider',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [MAT_SLIDER_VALUE_ACCESSOR],
+})
+export class MatSlider implements AfterViewInit, OnChanges, OnDestroy, ControlValueAccessor {
+  /** Event emitted when the slider value has changed. */
+  @Output() readonly change: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
+
+  /** Event emitted when the slider thumb moves. */
+  @Output() readonly input: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
+
+  /**
+   * Emits when the raw value of the slider changes. This is here primarily
+   * to facilitate the two-way binding for the `value` input.
+   * @docs-private
+   */
+  @Output() readonly valueChange: EventEmitter<number> = new EventEmitter<number>();
+
+  /** Tabindex for the slider. */
+  @Input() tabIndex: number = 0;
+
+  /** The color palette for this slider. */
+  @Input() color: ThemePalette = 'accent';
+
+  /**
+   * Function that will be used to format the value before it is displayed
+   * in the thumb label. Can be used to format very large number in order
+   * for them to fit into the slider thumb.
+   */
+  @Input() displayWith: (value: number) => string | number;
+
+  /** The minimum value that the slider can have. */
+  @Input()
+  get min(): number {
```

Comment:
Yep, I usually do this as well, but clang-format formatted it like that for me 🤷‍♂ 

I guess I'll keep it like that as otherwise I will hit it the next time when I run clang-format.